skills/dff_short_story_skill/scenario/response.py

```diff
@@ -101,16 +116,99 @@ def fallback(ctx: Context, actor: Actor, *args, **kwargs) -> str:
-        int_ctx.set_can_continue(ctx, actor, "CANNOT_CONTINUE")
+        int_ctx.set_can_continue(ctx, actor, CAN_NOT_CONTINUE)
         return "Oh, sorry, but I've run out of stories."
 
     # no stories
     elif prev_node == "which_story_node" and not story_type:
-        int_ctx.set_can_continue(ctx, actor, "CAN_CONTINUE")
+        int_ctx.set_can_continue(ctx, actor, CAN_CONTINUE_SCENARIO)
         return random.choice(sorted(phrases.get("no_stories", [])))
 
     # if prev_node is tell_punchline_node or fallback_node
     else:
-        int_ctx.set_can_continue(ctx, actor, "MUST_CONTINUE")
+        int_ctx.set_can_continue(ctx, actor, MUST_CONTINUE)
         int_ctx.set_confidence(ctx, actor, 0.5) if int_cnd.is_do_not_know_vars(ctx, actor) else None
         return random.choice(sorted(phrases.get("start_phrases", [])))
+
+
+def generate_story(ctx: Context, actor: Actor, *args, **kwargs) -> str:
+    reply = ""
+    utt = int_ctx.get_last_human_utterance(ctx, actor)["text"]
+    if utt:
+        full_ctx = int_ctx.get_human_utterances(ctx, actor)
+        nouns = full_ctx[-1].get("annotations", {}).get("rake_keywords", [])
+        if len(full_ctx) > 1:
+            nouns_tmp = full_ctx[-2].get("annotations", {}).get("rake_keywords", [])
+            nouns_tmp.extend(nouns)
+            nouns = nouns_tmp
+        logger.info(f"Keywords from annotator: {nouns}")
+        try:
+            resp = requests.post(STORYGPT_KEYWORDS_SERVICE_URL, json={"utterances_histories": [[nouns]]}, timeout=2)
+            raw_responses = resp.json()
+            int_ctx.set_confidence(ctx, actor, 0.9)
+            int_ctx.set_can_continue(ctx, actor, CAN_CONTINUE_SCENARIO)
+        except Exception as exc:
+            logger.exception(exc)
+            sentry_sdk.capture_exception(exc)
+            int_ctx.set_confidence(ctx, actor, 0.0)
+            int_ctx.set_can_continue(ctx, actor, CAN_CONTINUE_SCENARIO)
+            return ""
+        reply = raw_responses[0][0]
+        reply = "Oh, that reminded me of a story! " + reply
+    else:
+        int_ctx.set_confidence(ctx, actor, 0.0)
+        int_ctx.set_can_continue(ctx, actor, CAN_CONTINUE_SCENARIO)
+        logger.info("No context")
+    return reply
+
+
+def choose_noun(nouns):
+    for noun in nouns:
+        story_word = re.search(story_pattern, noun)
+        if not story_word:
+            return noun
+    return ""
+
+
+def choose_topic(ctx: Context, actor: Actor, *args, **kwargs) -> str:
+    int_ctx.set_can_continue(ctx, actor, MUST_CONTINUE)
+    int_ctx.set_confidence(ctx, actor, 1.0)
+    return random.choice(STORY_TOPIC_QUESTIONS)
+
+
+def generate_prompt_story(ctx: Context, actor: Actor, *args, **kwargs) -> str:
+    utt = int_ctx.get_last_human_utterance(ctx, actor)
+    last_utt = utt["text"]
+    logger.info(f"Utterance: {last_utt}")
+    if last_utt:
+        nouns = utt.get("annotations", {}).get("spacy_nounphrases", [])
```

Comment:
get_entities from common.utils
not only named & without types
it will be better way to get nouns